inst/templates/ad_advs.R

```diff
@@ -0,0 +1,452 @@
+    new_vars = adsl_vars,
+    by_vars = get_admiral_option("subject_keys")
+  ) %>%
+  ## Calculate BRTHDT ----
+  derive_vars_dt(
+    new_vars_prefix = "BRTH",
+    dtc = BRTHDTC
+  ) %>%
+  ## Calculate ADT, ADY ----
+  derive_vars_dt(
+    new_vars_prefix = "A",
+    dtc = VSDTC
+  ) %>%
+  derive_vars_dy(reference_date = TRTSDT, source_vars = exprs(ADT)) %>%
+  ## Calculate Current Analysis Age AAGECUR and unit AAGECURU ----
+  derive_vars_duration(
+    new_var = AAGECUR,
+    new_var_unit = AAGECURU,
+    start_date = BRTHDT,
+    end_date = ADT
+  )
+
+advs <- advs %>%
+  ## Add PARAMCD only - add PARAM etc later ----
+  derive_vars_merged_lookup(
+    dataset_add = param_lookup %>% filter(!is.na(VSTESTCD)),
+    new_vars = exprs(PARAMCD),
+    by_vars = exprs(VSTESTCD)
+  ) %>%
+  ## Calculate AVAL ----
+  mutate(AVAL = VSSTRESN)
+
+## Get visit info ----
+# See also the "Visit and Period Variables" vignette
+# (https://pharmaverse.github.io/admiral/articles/visits_periods.html#visits)
+advs <- advs %>%
+  # Derive Timing
+  mutate(
+    ATPTN = VSTPTNUM,
+    ATPT = VSTPT,
+    AVISIT = case_when(
+      str_detect(VISIT, "UNSCHED|RETRIEVAL|AMBUL") ~ NA_character_,
+      !is.na(VISIT) ~ str_to_title(VISIT),
+      TRUE ~ NA_character_
+    ),
+    AVISITN = as.numeric(case_when(
+      VISIT == "SCREENING 1" ~ "-1",
+      VISIT == "BASELINE" ~ "0",
+      str_detect(VISIT, "WEEK") ~ str_trim(str_replace(VISIT, "WEEK", "")),
+      TRUE ~ NA_character_
+    ))
+  )
+
+# Derive Current HEIGHT/LENGTH at each time point Temporary variable----
+advs <- advs %>%
+  derive_vars_merged(
+    dataset_add = advs,
+    by_vars = c(get_admiral_option("subject_keys"), exprs(AVISIT)),
+    filter_add = PARAMCD == "HEIGHT" & toupper(VSSTRESU) == "CM",
+    new_vars = exprs(HGTTMP = AVAL, HGTTMPU = VSSTRESU)
+  )
+
+# Merge ADVS to the chosen Growth metadata as an input to meta_criteria ----
+## Derive Anthropometric indicators (Z-Scores/Percentiles-for-Age) based on Standard Growth Charts ----
+## For Height/Weight/BMI/Head Circumference by Age ----
+advs_age <- advs %>%
+  derive_params_growth_age(
+    sex = SEX,
+    age = AAGECUR,
+    age_unit = AAGECURU,
+    meta_criteria = weight_for_age,
+    parameter = VSTESTCD == "WEIGHT",
+    analysis_var = AVAL,
+    set_values_to_sds = exprs(
+      PARAMCD = "WGTASDS",
+      PARAM = "Weight-for-age z-score"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "WGTAPCTL",
+      PARAM = "Weight-for-age percentile"
+    )
+  ) %>%
+  derive_params_growth_age(
+    sex = SEX,
+    age = AAGECUR,
+    age_unit = AAGECURU,
+    meta_criteria = height_for_age,
+    parameter = VSTESTCD == "HEIGHT",
+    analysis_var = AVAL,
+    set_values_to_sds = exprs(
+      PARAMCD = "HGTSDS",
+      PARAM = "Height-for-age z-score"
+    ),
+    set_values_to_pctl = exprs(
+      PARAMCD = "HGTPCTL",
+      PARAM = "Height-for-age percentile"
+    )
+  ) %>%
+  derive_params_growth_age(
```

Comment:
For BMI only i think we should set `bmi_cdc_correction = TRUE` as David had advised us that it is appropriate to use the CDC developed extended percentiles (>95%) as default to monitor high BMI values

Reply:
@Lina2689 FYI to update this one too

Reply:
@zdz2101 @rossfarrugia I set `bmi_cdc_correction = TRUE` for BMI, but its throwing an error for P95 and sigma. Can you please suggest from I could get this?

<img width="497" alt="image" src="https://github.com/pharmaverse/admiralpeds/assets/157117024/c4a50ab9-bf9e-4275-b66a-5ab51a7b9f9f">

Reply:
@Lina2689 are you sure you're using the right source metadata? maybe you didn't merge the latest `main` version of https://github.com/pharmaverse/admiralpeds/blob/main/data-raw/cdc_bmi.R here. if you look at that file it does have those columns. or if you spot an issue here with `data/cdc_bmiage.rda` let us know.

Reply:
maybe your `bmi_for_age` step of preprocessing is inadvertently dropping these variables.

Reply:
Thanks @rossfarrugia I missed to refresh the cdc_bmiage.rda file.